packages/melos/lib/src/package.dart

```diff
@@ -596,6 +597,28 @@ The packages that caused the problem are:
     return PackageMap(packageMap, logger);
   }
 
+  /// Combines a [PackageMap] with a root package if the root package 
+  /// name doesn't conflict with existing packages.
+  // ignore: prefer_constructors_over_static_methods
```

Comment:
I think this method naming is consistent (by analogy with `resolveRootPackage`), that’s why I added the ignore here. I can update it if you suggest any better alternatives.

Reply:
Can't it just be `addPackage` on the instance instead? Then it doesn't need to be `var` at the other place either.